extension/chrome/elements/compose-modules/compose-storage-module.ts

```diff
@@ -102,20 +102,35 @@ export class ComposeStorageModule extends ViewModule<ComposeView> {
     try {
       const lookupResult = await this.view.pubLookup.lookupEmail(email);
       if (lookupResult && email) {
-        if (lookupResult.pubkey) {
-          const key = await KeyUtil.parse(lookupResult.pubkey);
-          if (!key.usableForEncryption && !KeyUtil.expired(key)) { // Not to skip expired keys
+        const pubkeys: Key[] = [];
+        for (const pubkey of lookupResult.pubkeys) {
+          const key = await KeyUtil.parse(pubkey);
+          if (!key.usableForEncryption && !key.revoked && !KeyUtil.expired(key)) { // Not to skip expired and revoked keys
             console.info('Dropping found+parsed key because getEncryptionKeyPacket===null', { for: email, fingerprint: key.id });
             Ui.toast(`Public Key retrieved for email ${email} with id ${key.id} was ignored because it's not usable for encryption.`, 5);
-            lookupResult.pubkey = null; // tslint:disable-line:no-null-keyword
+          } else {
+            pubkeys.push(key);
           }
         }
-        const ksContact = await ContactStore.obj({ email, name, pubkey: lookupResult.pubkey, lastCheck: Date.now() });
-        if (ksContact.pubkey) {
-          this.ksLookupsByEmail[email] = ksContact.pubkey;
+        // save multiple pubkeys as separate operations
+        // todo: add a convenient method to storage?
+        const updates: ContactUpdate[] = [];
+        if (!pubkeys.length) {
+          if (name) {
+            // update just name
+            updates.push({ name } as ContactUpdate);
+          } else {
+            return ContactStore.obj({ email });
```

Comment:
Do we need to save this object (containing only the `email` key) to the store?
`name` is null, `pubkey` is null, lastCheck is now set per pubkey, so it can't be set either.

Reply:
There is benefit to having email addresses of contacts in the local store even if no public key is associated with them. When looking up contacts next time, these contacts will show up in gray and I may want to still send email to them (password encrypted instead of public key encrypted).

If at the point this line executes the contact may email not be saved into storage yet, then it's good to save it.

If at this point the contact is always already stored in storage (because whichever method calls this always already did it), then you could skip this.